File: backend/app/Enums/EventCategory.php
<?php

namespace App\Enums;

enum EventCategory: string
{
    case Workshop = 'Workshop';
    case Meetup = 'Meetup';
    case Academic = 'Academic';
    case Sports = 'Sports';
    case Club = 'Club';
    case Career = 'Career';
    case Social = 'Social';
    case General = 'General';

    /**
     * @return list<string>
     */
    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}

File: mobile/lib/modules/events/models/event_category.dart
/// Allowed event categories. Keep in sync with `App\Enums\EventCategory`.
abstract final class EventCategory {
  static const workshop = 'Workshop';
  static const meetup = 'Meetup';
  static const academic = 'Academic';
  static const sports = 'Sports';
  static const club = 'Club';
  static const career = 'Career';
  static const social = 'Social';
  static const general = 'General';

  static const values = [
    workshop,
    meetup,
    academic,
    sports,
    club,
    career,
    social,
    general,
  ];
}
